File: python/breadth-first/sample1.py
from collections import deque


graph = {}
graph["you"] = ["alice", "bob", "claire"]
graph["bob"] = ["anuj", "peggy"]
graph["alice"] = ["peggy"]
graph["claire"] = ["thom", "jonny"]
graph["anuj"] = []
graph["peggy"] = []
graph["thom"] = []
graph["jonny"] = []
graph["tomm"] = []

def person_is_seller(name):
	print(name[-1])
	return name[-1] == 'm'

def search(name):
	search_queue = deque()
	search_queue += graph[name]
	searched = [] #this array keeps track of which people you've searched before
	while search_queue:
		person = search_queue.popleft()
		print('next queue in search: '+person)
		if not person in searched: #only search this person if you haven't already searched them
			if person_is_seller(person):
				print (person + " is a mango seller!")
				return True
			else:
				print (graph[person])
				print('==people remaining in search queue==')
				search_queue += graph[person]
				print(search_queue)
				print('==end==')
				searched.append(person) #marks this person as searched
		else:
			print("note: " + person + " is already searched")
	return False

print (search("you"))

File: javascript/breadth-first/sample1.js
var graph = {}
graph["you"] = ["alice", "bob", "claire"]
graph["bob"] = ["anuj", "peggy"]
graph["alice"] = ["peggy"]
graph["claire"] = ["thom", "jonny"]
graph["anuj"] = []
graph["peggy"] = []
graph["thom"] = []
graph["jonny"] = []
graph["tomm"] = []

// console.log(graph);
// var test = "name";
// console.log(test[test.length-1]);

function person_is_seller(name) {
	return name[name.length-1] === 'm';
}

function search(name) {
	var search_queue = [];
	var searched = [];
	search_queue = search_queue.concat(graph[name]);
	while (search_queue.length > 0) {
		var person = search_queue.shift();

		/*Sonny's note: Create a node-like relationship to each graph using D3.js demonstrating each relationship. */
		/*hint: Calculate the distance between 1st-rate relationships to 2nd-rate, so on and so forth. If possible, put an arrow for each connection*/
		// var para = document.createElement("div");
		// para.className = "node-circle";
		// var node = document.createTextNode(person);
		// para.appendChild(node);

		// var elem = document.getElementById("div1");
		// elem.appendChild(para);
		/*END*/

		if(searched.indexOf(person) < 0) {
			// console.log('no searched person');
			if (person_is_seller(person)) {
				console.log(person + " is a mango seller!");
				return true;
			}
			else {
				search_queue = search_queue.concat(graph[person]);
				searched.push(person);
			}
		}
		console.log(person);
	}
	
}

search("you");